### PR TITLE
[MM-17353] Wait for transition on RHS before rendering expensive sub components

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
When the `search_results` component contained a large number of results, rendering would lag the transition on tablet view, which is not ideal from a UX standpoint. This PR stops the results from being rendered until the transition completes, improving visual performance.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-17353

### CODE REVIEW

File: components/sidebar_right/sidebar_right.jsx [hunk 1]
```diff
@@ -55,6 +76,27 @@ export default class SidebarRight extends React.PureComponent {
         }
     }
 
+    determineTransition = () => {
+        var transitionInfo = window.getComputedStyle(this.sidebarRight.current).getPropertyValue('transition');
+        var hasTransition = transitionInfo !== 'all 0s ease 0s';
```
Comment (jwilander) [inline components/sidebar_right/sidebar_right.jsx]: nit: Generally we try to use `const` or `let` as appropriate instead of `var`

Comment (devinbinnie) [reply components/sidebar_right/sidebar_right.jsx]: Whoops. Old habits die hard :P

**Comment by lindalumitchell** [comment]
Just took a look, and wonder if what I'm seeing is an issue: If the at-mention RHS is already open when change the screen width, the RHS can change from showing results to showing an endless loading indicator:

mentions-not-loading-narrowed-screen 2019-09-05 01_23_08

**Comment by devinbinnie** [comment]
@lindalumitchell Yep that's definitely not ideal, I'll push up a fix.
@jasonblais I think we're okay, I've scoped this to only work with the `search_results` component so that it should only affect what it needs to, and resizes are handled (apart from the above issue). Not sure if anyone else has any cases worth testing before this goes in.

**Comment by levb** [comment]
@lindalumitchell not merging until you re-test and remove `QA review`?

**Comment by devinbinnie** [comment]
@amyblais can this going into v5.15?

**Comment by amyblais** [comment]
Sure, is relatively non-risky?

**Comment by devinbinnie** [comment]
> Sure, is this relatively non-risky?

Relatively, the worst case scenario is the search results don't load on a certain view, but we've covered all cases. Just waiting for QA to suggest any additional cases to test before this goes in.

**Comment by lindalumitchell** [comment]
Thanks @devinbinnie, I've done some more testing on the new build of the test server, and found no issues. Some areas I tested, for reference: 
- Wide, mid (tablet), and narrow (mobile), and transitions between them for:
- RHS (at-mentions, flags, pinned posts, search results), with: 
- returned results and with empty results, 
- pinned posts when switching channels, 
- RHS expanded while transitioning

Removing QA Review label. Thanks!

**Pull request merged**